cloud/src/contexts/project.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  type ReactNode,
+} from "react";
+import type { PublicProject } from "@/db/schema";
+import { useProjects } from "@/src/api/projects";
+import { useOrganization } from "@/src/contexts/organization";
+
+const STORAGE_KEY = "mirascope:selectedProjectId";
+
+type ProjectContextType = {
+  projects: readonly PublicProject[];
+  selectedProject: PublicProject | null;
+  setSelectedProject: (project: PublicProject | null) => void;
+  isLoading: boolean;
+};
+
+const ProjectContext = createContext<ProjectContextType | null>(null);
+
+export function ProjectProvider({ children }: { children: ReactNode }) {
+  const { selectedOrganization } = useOrganization();
+  const { data: projects = [], isLoading } = useProjects(
+    selectedOrganization?.id ?? null,
+  );
+  const [selectedProject, setSelectedProjectState] =
+    useState<PublicProject | null>(null);
+
+  // Load selected project from localStorage on mount or when org changes
+  useEffect(() => {
+    const storedId = localStorage.getItem(STORAGE_KEY);
+    if (storedId && projects.length > 0) {
+      const project = projects.find((p) => p.id === storedId);
+      if (project) {
+        setSelectedProjectState(project);
+      } else {
+        // If stored project doesn't exist in current org, select first one
+        setSelectedProjectState(projects[0]);
+        localStorage.setItem(STORAGE_KEY, projects[0].id);
+      }
+    } else if (projects.length > 0 && !selectedProject) {
+      // Auto-select first project if none selected
+      setSelectedProjectState(projects[0]);
+      localStorage.setItem(STORAGE_KEY, projects[0].id);
+    } else if (projects.length === 0) {
+      // Clear selection if no projects
+      setSelectedProjectState(null);
```

Comment:
As above: Let's use setSelectedProject instead of re-implementing the logic here